jenkins-plugin/src/test/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepTest.java

```diff
@@ -96,6 +106,47 @@ public void testDisableAllPublishers() throws Exception {
         assertFingerprintDoesNotExist(COMMONS_LANG3_FINGERPRINT);
     }
 
+    // the jdk path is configured in Dockerfile
+    private static Stream<Arguments> jdkMapProvider() {
+        return Stream.of(
+                arguments("jdk8", "/opt/java/jdk8"),
+                arguments("jdk11", "/opt/java/jdk11")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("jdkMapProvider")
+    @Issue("JENKINS-71949")
+    public void tesWithDifferentJavasForBuild(String jdkName, String jdkPath) throws Exception {
+        loadMonoDependencyMavenProjectInGitRepo(this.gitRepoRule);
+        String gitRepoPath = this.gitRepoRule.toString();
+        javaGitContainerRule.copyFileToContainer(MountableFile.forHostPath(gitRepoPath), "/tmp/gitrepo");
+        javaGitContainerRule.execInContainer("chmod", "-R", "777", "/tmp/gitrepo");
+        registerAgentForContainer(javaGitContainerRule);
+        ToolLocationNodeProperty.ToolLocation toolLocation =
+                new ToolLocationNodeProperty.ToolLocation(new JDK.DescriptorImpl(), jdkName, jdkPath);
+        ToolLocationNodeProperty toolLocationNodeProperty = new ToolLocationNodeProperty(toolLocation);
+        Objects.requireNonNull(jenkinsRule.jenkins.getNode(AGENT_NAME)).getNodeProperties().add(toolLocationNodeProperty);
+
+        jenkinsRule.jenkins.getJDKs().add(new JDK(jdkName, jdkPath));
+
+        //@formatter:off
+        WorkflowRun run = runPipeline(Result.SUCCESS,
+                "node('" + AGENT_NAME + "') {\n" +
+                "  sh 'ls -alrt /tmp'\n" +
+                "  git('/tmp/gitrepo')\n" +
+                "  withMaven(jdk: '" + jdkName + "') {\n" +
```

Comment:
you need to add `traceability: true`  option to get the logs `artifactsPublisher - ...`

Reply:
not sure to understand your comment?
The code here https://github.com/jenkinsci/pipeline-maven-plugin/blob/003efafd6e818ef9f964dc343c323751d010ca1f/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/GeneratedArtifactsPublisher.java#L105 doesn't any flag activated to be logged.

Reply:
hum yes sorry, test is failing on my side because there is no ` artifactsPublisher` logs

because of the ` package`  target of the maven invocation

Reply:
anyway the most important assert is the assert with the ArtifactManager

Reply:
> hum yes sorry, test is failing on my side because there is no ` artifactsPublisher` logs
> 
> because of the ` package` target of the maven invocation

? can you explain because the pipeline is using `package` and as far as I can see the test works (locally or in the CI)

Reply:
sorry, short and bad night, I just need coffee ... I mix between `artifactsPublisher` and `pipelineGraphPublisher`

Test was red on my side because I forgot to change maven-spy pom :man_facepalming: